spec.html

```diff
@@ -16573,10 +16570,13 @@ <h2>Syntax</h2>
           DecimalFraction[~Empty]
 
         DecimalIntegerLiteral ::
+          CanonicalDecimalIntegerLiteral
+          NonOctalDecimalIntegerLiteral
+
+        CanonicalDecimalIntegerLiteral ::
```

Comment:
Might there be a better name than CanonicalDecimalIntegerLiteral? It makes me think “canonical numeric string”, which corresponds to neither a superset nor subset of the sequences this production matches.

Reply:
Sure, suggestions welcome. But the similarity to "canonical numeric string" is intentional, and the two concepts do mostly agree where they intersect—not for power-of-two numbers large enough that [Number::toString](https://tc39.es/ecma262/multipage/ecmascript-data-types-and-values.html#sec-numeric-types-number-tostring) uses exponential notation, but I'm willing to chalk that up as a difference between canonicalized numeric strings vs. integer literals (numeric strings and integer literals already being distinct with respect to e.g. supporting `_` separators).

Reply:
> not for power-of-two numbers large enough that [Number::toString](https://tc39.es/ecma262/multipage/ecmascript-data-types-and-values.html#sec-numeric-types-number-tostring) uses exponential notation

Even without exponential notation, stuff beyond safe integer range won’t always be canonical in the CNS sense, right? For example the source text literal `9007199254740993` evaluates to the Number value whose canonical string representation is `9007199254740992`.

> the similarity to "canonical numeric string" is intentional

In that case, perhaps I’m alone in thinking the word might be misleading there, so unless others have an opinion it’s probably not worth changing.